File: src/classes/device_ip_info.rs
use super::addresses::{ipv4::IPv4Addr, mac_address::MacAddress};

#[derive(Debug, Clone)]
pub struct DeviceIPInfo
{
   device_id: i64,
   device_mac: MacAddress,
   local_ip_address: IPv4Addr
}

impl DeviceIPInfo{
    pub fn new(device_id: i64, device_mac: MacAddress, local_ip_address: IPv4Addr) -> Self{
        Self{
            device_id,
            device_mac,
            local_ip_address,
        }
    }

    pub fn get_device_id_ref(&self) -> &i64 {
        &self.device_id
    }
    
    pub fn get_device_id_mut(&mut self) -> &mut i64 {
        &mut self.device_id
    }
    
    pub fn get_device_mac_ref(&self) -> &MacAddress {
        &self.device_mac
    }
    
    pub fn get_device_mac_mut(&mut self) -> &mut MacAddress {
        &mut self.device_mac
    }
    
    pub fn get_local_ip_address_ref(&self) -> &IPv4Addr {
        &self.local_ip_address
    }
    
    pub fn get_local_ip_address_mut(&mut self) -> &mut IPv4Addr {
        &mut self.local_ip_address
    }
}

File: lib/classes/device_ip_info.py
class DeviceIPInfo:
    def __init__(self, device_id: int, device_mac: str, local_ip_address: str):
        self.device_id = device_id
        self.device_mac = device_mac
        self.local_ip_address = local_ip_address

    def get_device_id(self):
        return self.device_id
    def get_device_mac(self):
        return self.device_mac
    def get_local_ip_address(self):
        return self.local_ip_address
    
    def set_device_id(self, device_id):
        self.device_id = device_id
    def set_device_mac(self, device_mac):
        self.device_mac = device_mac
    def set_local_ip_address(self, local_ip_address):
        self.local_ip_address = local_ip_address
    